feat(bixarena): add a set of example prompts for release (SMR-578, SMR-579)

rrchai opened the pull request:
## Description

Add a set of example prompts for release and make the entire surface of the example prompts boxes clickable

Used custom HTML buttons instead of Gradio buttons because Gradio Button doesn't have child div which makes it impossible to have both full surface clickable container and CSS line-clamp in the same time.

## Related Issue

[SMR-578](https://sagebionetworks.jira.com/browse/SMR-578)
[SMR-579](https://sagebionetworks.jira.com/browse/SMR-579)

## Changelog

- Add a set of example prompts for release (31)
- Add clickable example prompt cards with horizontal navigation (left/right arrows)
- Remove the `refresh_prompts()` calls from the navigation button handlers 

## Preview

<img width="1115" height="468" alt="Screen Shot 2025-11-01 at 12 55 06 AM" src="https://github.com/user-attachments/assets/4d5997a9-5e4e-41fc-a128-5be78e395f5a" />



[SMR-578]: https://sagebionetworks.jira.com/browse/SMR-578?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[SMR-579]: https://sagebionetworks.jira.com/browse/SMR-579?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

rrchai commented:
It looks like the upstream introduces an unexpected behavior that clicking "Battle" button will create the example prompt cards during a battle:

![chrome-capture-2025-10-31](https://github.com/user-attachments/assets/6f59d1b9-b289-48c4-8269-1e41f0f61420)

I will see if I can fix it in this PR.

tschaffter commented on apps/bixarena/api/src/main/resources/db/migration/V1.0.2__insert_example_prompts.sql:
Some prompts are so long that they don't fit into their box. The example prompt should be entirely visible in the box and preferably of reasonable length so that the user can read them quickly.

Suggestion:
1. Identify the max length of the text that could fit in the box + margin of error because of how the text is wrapped at the word level.
2. Identify PubMedQA prompts that meet this max length requirements and pic 20-30 of them.
3. Set max length constraints to the OpenAPI spec and SQL table definition.
4. (Optional) The boxes could be made a tiny bit wider if deemed necessary. The text should ideally not spread over more than 3-4 lines.

tschaffter reviewed (changes requested):
Clicking anywhere on the boxes to start a battle works nicely! I have one request regarding the length of the example prompts.

rrchai commented:
> 1. Identify the max length of the text that could fit in the box + margin of error because of how the text is wrapped at the word level.
> 2. Identify PubMedQA prompts that meet this max length requirements and pic 20-30 of them.
> 3. Set max length constraints to the OpenAPI spec and SQL table definition.
> 4. (Optional) The boxes could be made a tiny bit wider if deemed necessary. The text should ideally not spread over more than 3-4 lines.

I’ve widened the boxes and kept the text clamped at three lines. From testing, each box fits around 170 -180 characters. Looking at the distribution, most of examples we have stay under 160 characters. I tried to increase the line clamp to 4 lines to cover 200 characters, but that would create noticeable empty space for shorter one-line examples. Then I take a safer ~10% error margin, so it rounds up to 160 characters as the maximum length.

<img width="3568" height="1768" alt="prompt_length_distribution" src="https://github.com/user-attachments/assets/e58f6ddb-1940-4578-90b3-cb8f3bbbf3da" />

I’ve updated the SQL table to include only the records that meet the defined maximum length constraint. I also added additional examples from PubMedQA, resulting in a total of 20 examples: 16 from PubMedQA and 4 from Bixarena (custom examples contributed by the team).

tschaffter reviewed (approved):
The example prompt cards are now fully clickable, and I’ve generated a wide range of random example prompts — all of which fit nicely within the card layout. Thanks!
